Casecommons merge

brysgo opened the pull request:
Since it is widely agreed that the Casecommons fork has been better maintained up until now I went through as carefully as I could and tried to pull in all the changes they've made. I would like a few people to try it out before I merge it in.

Commit:
Added a README

Commit:
Edited README.md via GitHub

Commit:
increase the max file size that CommandT handles

Commit:
Add MIT License

Commit:
Merge pull request #11 from ragaskar/add-mit-license

Add MIT License

Commit:
Pull in some keybindings from Pivotal's config.

* Auto-indent whole file
* Jump to new line in insert mode

Commit:
Remove PeepOpen.

No one uses it, and the mapping is already overridden later in the
config.

Commit:
Merge pull request #12 from Peeja/pull/cc/readme

Add README.

Commit:
Merge pull request #13 from Peeja/pull/cc/reformat-mapping

Pull in some keybindings from Pivotal's config.

Commit:
Merge pull request #14 from Peeja/pull/both/peepopen

Remove PeepOpen.

Commit:
Merge pull request #10 from sumancc/increase_max_file_size

increase the max file size that CommandT handles

joemoore committed:
turn off the blinking cursor, as it redraws the buffer. This can be slow in some remote pairing situations.

joemoore committed:
Removing vim.vim, moving cursor options to options.vim

Commit:
Merge pull request #15 from joemoore/master

turn off the blinking cursor, as it redraws the buffer.  This can be slow in some remote pairing situations.

Commit:
switched command-t's url from wincent to github

Commit:
Merge pull request #16 from mikfreedman/master

switched command-t's url from wincent to github

had problems cloning the vim config this week due to command-t's git repo

Commit:
remove autoclose as per Feb 28 standup

jgeiger committed:
Add autocorrect for common spelling errors

Commit:
Merge pull request #18 from jgeiger/master

Add autocorrect to vim config

Commit:
Fix lack of help files for plugins

add plugin dirs to path before loading help

pivotal-casebook committed:
Merge pull request #19 from camelpunch/fix_help

Fix help

jgeiger committed:
Typing quickly, I end up hitting "W" a lot which isn't bound and just makes me retype "w" to save

jgeiger committed:
Add command for W

jgeiger committed:
add pastetoggle keybinding

pivotal-casebook committed:
Merge pull request #17 from camelpunch/master

Remove the autoclose plugin

Commit:
Add the candycode color scheme

Commit:
Add syntastic plugin

pivotal-casebook committed:
Merge pull request #22 from matt-royal/master

Add syntax error highlighting and a new color scheme (candycode)

Commit:
Use bundle exec for ctags

Commit:
Enable folds and operator highlighting in Ruby files

Add an autocmd to prevent files from opening with all folds closed

pivotal-casebook committed:
Merge pull request #23 from matt-royal/master

Enable folding and operator highlighting in Ruby files

Commit:
Change ,t and ,T to only build tags for app code (no gems, no specs)

Commit:
Generalize ,t and ,T to work in any Rails project

Commit:
Disable folding in Ruby due to performance issues

davidxia committed:
Fixed README installation instructions to have correct github repo url

Commit:
Add "bundle exec" to ,T

Commit:
,T now includes tags for javascript files

Commit:
Improve CB2 abbreviations.

Commit:
Add abbreviation RL:: => RemovalLocation

Commit:
Add DiffThese command, which makes diffing easier

Commit:
Improve DiffThese - undiffed windows now scroll with the diffed windows

Commit:
DiffThese defaults to diffing the top 3 windows (assuming vimdiff)

Commit:
Add DiffTheirs,DiffMine,DiffChanges,DiffAll,DiffOff commands

Commit:
Fix DiffOff

Commit:
Rename DiffTheirs & DiffMine to DiffLeft & DiffRight, and add DiffWith

Commit:
Add Tmux/Vim Autowrite support.

Commit:
Make vim in tmux act more better.

(Act like MVim)

pivotal-casebook committed:
Merge pull request #24 from davidxia/readme-fix

Fixed README installation instructions to have correct github repo url

Commit:
added vim colors sample pack

Commit:
Add handlebars syntax

Commit:
Add edit-plus vim plugin

Commit:
added vimlog to gitignore

Commit:
added kj->command mode keybinding

Commit:
Revert "added kj->command mode keybinding"

This reverts commit eda08800ff35dad1b0ce97fd6924e20b2b17fcc5.

Commit:
Add EnhancedJumps so we can jump forward/back a file at a time

Commit:
Add gems.tags to ctags list

This allows for good integration with
guard-ctags-bundler

Commit:
run tests from vim with test_server plugin

Commit:
Revert "run tests from vim with test_server plugin"

This reverts commit 3310941e417419c65e3a4221d579ce3d14c3b9a6.

Commit:
Add back the test_server, now with the .gitsubmodule file set up correctly

Commit:
Add default ackrc and instructions for installation

pivotal-casebook committed:
Merge pull request #29 from pivotal-medici/ack_config

Ack config

pivotal-casebook committed:
Merge pull request #25 from jgeiger/a2bedb0b

Add W as alias for w, set <f2> to toggle paste mode

Commit:
Use the diffthese as a submodule

nertzy committed:
Merge pull request #30 from pivotal-medici/handlebars_highlighting

Add handlebars syntax

Commit:
Add tomorrow theme

Commit:
Bump test_server to new version

Commit:
Bump vim-rails

Commit:
Bump nerdtree

Commit:
Bump vim-ruby

Commit:
Bump some themes

Commit:
bump gundo

Commit:
Bump fugitive

Commit:
Bump unimpaired

Commit:
Bump command-t

Commit:
Bump ack

Commit:
Bump vim-javascript

Commit:
Bump vim-haml

Commit:
Bump vim-cucumber

Commit:
Bump syntastic

Commit:
Bump vim-ruby-refactoring

Commit:
Bump nerdcommenter

Commit:
Bump tabular

Commit:
Bump vim-abolish

Commit:
Bump vim-endwise

Commit:
Bump vim-repeat

Commit:
Bump vim-markdown

Commit:
Bump enhancedjumps

Commit:
Bump vim-slim

Commit:
Bump handlebars

Commit:
Bump textobj plugins

Commit:
bump vim-surround

Commit:
Add vim-powerline

Turn off insert_color_background_mode for now

Commit:
Switch to Tomorrow-Night color scheme by default

Commit:
Choose Menlo as the font

It looks better than Inconsolata for powerline

Commit:
Map test_server to <leader>t and remap generate ctags to <leader>rt

jgeiger committed:
Remove F12 keybinding since we're using it to run tests.

Commit:
removed enhanced jumps for badly behaving git repo

Commit:
remove ignorecase regex should work like normal

Commit:
Merge pull request #31 from jgeiger/master

Remove extra F12 keybinding since it's used for RunTest

Commit:
Bump syntastic to pick up bugfixes

Commit:
Don't unnecessarily use a private URL for a git submodule

Commit:
Add <leader>tt to re-run last test in test-server

Commit:
Add ag.vim for faster text searching via silver searcher

Commit:
Update vim plugins to match upstream master

Commit:
Added supertab

Commit:
Edit crontab in-place

Commit:
Add tmux-config

Commit:
Add vim-mustache

Commit:
Add vim-cucumber-align-pipes

Commit:
Don't manually set solarized_termcolors because it makes things worse

Commit:
Use unicode dot for trailing space in terminal vim

Commit:
add <leader>j for reverse J command

Commit:
Fix ESC in CommandT while in terminal Vim not working

pivotalcommon committed:
Merge pull request #1 from pivotalforks/updated_plugins

Updated plugins

pivotalcommon committed:
Merge pull request #2 from pivotalforks/add_ag_plugin

Add ag.vim for faster text searching via silver searcher

Commit:
Adds vim-coffee-script

Commit:
Adds jj exit imode binding

Commit:
Map <C-p> to CommandT

Commit:
Turn on ignorecase since smartcase has no effect without it

Commit:
Gracefully handle holding shift too long after : for common commands

Commit:
Add ZoomWin plugin

Commit:
Fixed timeout when using git protocol and gist

Commit:
remove vim-slim

Commit:
add new vim-slim

Commit:
new repo/new maintainer for vim-slim

Commit:
Use improved plugin to automatically close pairs and quotes.

Commit:
Don't consider ! or ? as part of a word in Ruby because it breaks things

See <https://github.com/tpope/vim-rails/issues/148#issuecomment-4644455>

Commit:
Bump a bunch of Vim plugins

Commit:
Bump nerdtree to latest version

Commit:
Add ag vim plugin

Need to `brew install the_silver_searcher` first

squeedee committed:
hammer time

pivotalcommon committed:
Merge pull request #6 from squeedee/master

Lets you preview all sorts of doc types.

Commit:
Add vim-blockle to toggle between {} and do/end for blocks

Commit:
Don't refresh buffer every time. Use <leader>F to refresh.

[Finishes #44470071]

Commit:
Add CC alias to show compiled coffeescript

Commit:
Add CamelCaseMotion plugin

Commit:
Move C-p mapping to keybindings file

Commit:
Remap FufBuffer to <leader>rb & remove redundant/unused Fuf mappings

pivotalcommon committed:
Update README.md to reflect move to pivotalcommon

Commit:
Add vim-gitgutter to display git changes in vim gutter

Commit:
Bump Vim plugins

ckundo committed:
Include rbenv instructions in README

Commit:
Remap vim-blockle to <leader>B

Commit:
Add tagbar plugin

Commit:
Update Pathogen

Commit:
Don't hide errors in .vimrc.local

Instead just don't load it if it doesn't exist, which is what I think
the original intent of `silent` here really was.

Commit:
Use edit-plus from Git

Commit:
Bump Vim plugins

Commit:
Hitting <tab> in insert mode should only insert a tab character

Commit:
We use Postgres, so set the default SQL dialect to Postgres

brysgo committed:
Merge pull request #8 from ckundo/readme

Include rbenv instructions in README

brysgo committed:
Carefully merge Casecommons/vim-config

brysgo commented:
Didn't mean to submit this to pivotalforks, before. @nertzy can you take a look at or try this branch when you get a chance?

nertzy commented:
Cool, I will give this a shot sometime soon, although I'm about to travel for a couple weeks so I might not be able to look at this too closely.

If there are too many conflicts, we could always throw the existing Pivotal config on a topic branch and force-push master to the same as Case Commons, but then I don't know if that would mess up some people in SF.

I bet your careful merge is good. I would try it on whichever project you're on for a week and look at the Vim logs to see what things look like after a couple days of solid use. Unfortunately we can't use the usual config at my current client since we're stuck on Windows and not everything is supported there.

brysgo committed:
Remove stray diff markup from readme

brysgo committed:
Remove random casebookisms identified by @nertzy

Commit:
Tweak to meet bertrands specifications

brysgo committed:
Revert to original ctags implementation

https://github.com/pivotalcommon/vim-config/commit/f397ca4544d8ec4a04ab420f279fadef0525a43e#commitcomment-3452069

Commit:
Add qargs plugin for global search/replace

brysgo committed:
Bump vim-coffee-script to fix missing reference

brysgo committed:
Remove after_bundles.vim

Hopefully it isnt important.

brysgo committed:
Remove auto-pairs plugin

I was wrong about the auto-matching functionality, it can use some work. I would be happy to except a more polished version. #5
